static/app/components/replays/videoReplayer.tsx

```diff
@@ -123,18 +119,48 @@ export class VideoReplayer {
         // previous video
         this.showVideo(el);
       }
-    });
+    };
 
-    el.addEventListener('loadedmetadata', event => {
+    const handlePlay = event => {
+      if (index === this._currentIndex) {
+        this._callbacks.onLoaded(event);
+      }
+    };
+
+    const handleLoadedMetaData = event => {
       // Only call this for current segment?
       if (index === this._currentIndex) {
+        // Theoretically we could have different orientations and they should
+        // only happen in different segments
         this._callbacks.onLoaded(event);
       }
+    };
+
+    el.addEventListener('ended', handleEnded);
+    el.addEventListener('loadeddata', handleLoadedData);
+    el.addEventListener('play', handlePlay);
+    el.addEventListener('loadedmetadata', handleLoadedMetaData);
+
+    this._listeners.push(() => {
+      el.removeEventListener('ended', handleEnded);
+      el.removeEventListener('loadeddata', handleLoadedData);
+      el.removeEventListener('play', handlePlay);
+      el.removeEventListener('loadedmetadata', handleLoadedMetaData);
     });
+  }
 
-    el.preload = 'auto';
+  private createVideo(segmentData: VideoEvent, index: number) {
+    const el = document.createElement('video');
+    el.src = `${this._videoApiPrefix}${segmentData.id}/`;
+    el.style.display = 'none';
+    el.setAttribute('type', 'video/mp4');
+    el.setAttribute('muted', '');
+    el.setAttribute('playinline', '');
+    el.setAttribute('preload', 'auto');
     // TODO: Timer needs to also account for playback speed
-    el.playbackRate = this.config.speed;
+    el.setAttribute('playbackRate', `${this.config.speed}`);
```

Comment:
does this do anything wrt the playback speed? it wasn't working before

Reply:
Yeah, that was the previous intent. (This *should* work, but it will be desynced from timer since timer won't match up with this playback rate)